spec/controllers/restaurants_controller_spec.rb

```diff
@@ -14,49 +14,45 @@
     it "creates a new restaurant" do
       post :create, params: { name: 'Artzieli11',
                                     cuisine: 'pizza',
-                                    rating: 3,
                                     ten_bis: true,
                                     address: 'street in tel aviv',
                                     max_delivery_time: 45 }
       expect(response.status).to eq(200)
-    end
-
-    it "can't create a restaurant with an invalid rating" do
-      post :create, params: { name: 'Artzieli',
-                              cuisine: 'pizza',
-                              rating: 7,
-                              ten_bis: true,
-                              address: 'street in tel aviv',
-                              max_delivery_time: 45 }
-      expect(response.status).to eq(400)
+      rest_id = JSON.parse(response.body)['id']
+      expect(Restaurant.count()).to eq(1)
```

Comment:
Notice that you are relying on the assumption that the table is empty before the test run

it is better to use this matcher in such cases
expect { some code }.to change{ SomeModel.count }.by(1)

Or if it is complicated,  save the initial number and then compare to it

Reply:
great advice, thanks!